packages/web-components/components/selfie/src/selfie-capture/SelfieCapture.js

```diff
@@ -816,9 +848,52 @@ class SelfieCaptureScreen extends HTMLElement {
     return this.hasAttribute('hide-attribution');
   }
 
-  get supportBothCaptureModes() {
-    const value = this.documentCaptureModes;
-    return value.includes('camera') && value.includes('upload');
+  async setupAgentMode() {
+    if (!this.allowAgentMode) {
+      return;
+    }
+
+    const supportAgentMode = await this.supportsAgentMode();
+
+    if (supportAgentMode || this.hasAttribute('show-agent-mode-for-tests')) {
+      this.switchCamera.hidden = false;
+    } else {
+      this.switchCamera.hidden = true;
+    }
+  }
+
+  async supportsAgentMode() {
+    try {
+      const devices = await navigator.mediaDevices.enumerateDevices();
+      const videoDevices = devices.filter(
+        (device) => device.kind === 'videoinput',
+      );
+
+      let hasBackCamera = false;
+
+      videoDevices.forEach((device) => {
+        // Check if the device label or device ID indicates a back camera
+        if (
+          device.label.toLowerCase().includes('back') ||
+          device.label.toLowerCase().includes('rear')
+        ) {
+          hasBackCamera = true;
+          return true;
+        }
+        return false;
+      });
+
+      if (hasBackCamera) {
+        console.warn('Device has a back camera.');
```

Comment:
i think that a) we shouldn't be console.warning something that is a successful operation and b) we should throw and handle (with a user facing message) when someone tries to use agent mode on a device which doesn't support it.

Reply:
removed